src/util/errors/ErrorUtil.ts

```diff
@@ -1,23 +1,30 @@
-import { types } from 'util';
 import { HttpError } from './HttpError';
 
 /**
  * Checks if the input is an {@link Error}.
  */
-export function isNativeError(error: any): error is Error {
-  return types.isNativeError(error);
+export function isError(error: any): error is Error {
+  const fields = [ 'name', 'message' ];
+  if (error.stack) {
+    fields.push('stack');
+  }
+  return fields.every((field: string): boolean => typeof error[field] === 'string');
```

Comment:
Maybe the generic is overkill here?

```suggestion
  return error &&
      typeof error.name === 'string' &&
      typeof error.message === 'string' &&
      (typeof error.stack ==='undefined' || typeof error.stack === 'string');
```

Reply:
And perhaps even with a short-circuit

```suggestion
  return error instanceof Error ||
      error &&
      typeof error.name === 'string' &&
      typeof error.message === 'string' &&
      (typeof error.stack ==='undefined' || typeof error.stack === 'string');
```